cmake/deprecated.cmake

```diff
@@ -0,0 +1,42 @@
+#######
+# Check for deprecated arguments, provide warning, and set the
+# corresponding option.
+#
+# This file is being added in support of https://github.com/Unidata/netcdf-c/pull/2895 and may eventually
+# no longer be necessary and removed in the future. 
+# 
+#######
+
+set(DEPR_OPT "Warning! Deprecated Options used. Please migrate your build system as follows:" CACHE INTERNAL "" FORCE)
```

Comment:
Similarly with the other comment, avoid using `set(CACHE)`. This is only to be used to expose for the user to overwrite or to save something across configure/build/install/test environment (but even there there are better ways).